library/alloc/src/rc.rs

```diff
@@ -1034,6 +1123,38 @@ impl<T: ?Sized> Rc<T> {
         inner
     }
 
+    /// Allocates an `RcBox<T>` with sufficient space for
+    /// a possibly-unsized inner value where the value has the layout provided,
+    /// returning an error if allocation fails.
+    ///
+    /// The function `mem_to_rcbox` is called with the data pointer
+    /// and must return back a (potentially fat)-pointer for the `RcBox<T>`.
+    unsafe fn try_allocate_for_layout(
```

Comment:
could you make `allocate_for_layout` reuse `try_allocate_for_layout(...).unwrap_or_else(...)`

Reply:
Done!

Reply:
oh ok. too bad the layout still needs to be recomputed.